app/src/main/java/com/android/partagix/ui/screens/ViewAccount.kt

```diff
@@ -0,0 +1,183 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import com.android.partagix.R
+import com.android.partagix.model.UserViewModel
+import com.android.partagix.ui.components.BottomNavigationBar
+import com.android.partagix.ui.navigation.NavigationActions
+import com.android.partagix.ui.navigation.Route
+import kotlin.math.round
+
+// @Preview(showBackground = true)
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ViewAccount(
+    modifier: Modifier = Modifier,
+    navigationActions: NavigationActions,
+    userViewModel: UserViewModel,
+) {
+  val uiState = userViewModel.uiState.collectAsState()
+  val user = uiState.value.user
+  Scaffold(
+      modifier = Modifier.fillMaxSize().testTag("viewAccount"),
+      topBar = {
+        TopAppBar(
+            title = { Text("My Account") },
+            modifier = Modifier.fillMaxWidth().testTag("title"),
+            navigationIcon = {
+              IconButton(onClick = { navigationActions.goBack() }) {
+                Icon(
+                    imageVector = Icons.AutoMirrored.Filled.ArrowBack,
+                    contentDescription = null,
+                    modifier = Modifier.width(48.dp))
+              }
+            })
+      },
+      bottomBar = {
+        BottomNavigationBar(
+            selectedDestination = Route.ACCOUNT,
+            navigateToTopLevelDestination = navigationActions::navigateTo,
+            modifier = modifier.testTag("accountScreenBottomNavBar"))
+      }) {
+        Column(
+            modifier =
+                Modifier.fillMaxHeight()
+                    .padding(it)
+                    .verticalScroll(rememberScrollState())
+                    .testTag("mainContent")) {
+              /*LaunchedEffect(key1 = uiState) {
+                  println("refreshed User: $user")
+                  user = uiState.user
+
+              }*/
+              Image(
+                  painter =
+                      painterResource(
+                          id = R.drawable.ic_launcher_background) /*TODO: get profile picture*/,
+                  contentDescription = null,
+                  modifier = Modifier.fillMaxWidth().testTag("userImage"),
+                  alignment = Alignment.Center)
+              Spacer(modifier = Modifier.height(8.dp))
+              Row(
+                  modifier = Modifier.fillMaxWidth().testTag("username"),
+                  horizontalArrangement = Arrangement.Absolute.SpaceAround) {
+                    val username = user.name
+                    Text("$username's profile")
+                  }
+              Spacer(modifier = Modifier.height(16.dp))
+
+              TextField(
```

Comment:
We can select (highlight it by clicking on it) this field making the user think it is editable -> maybe it should be not selectable.

Reply:
This has been addressed

Reply:
nice